src/async.c

```diff
@@ -808,21 +809,51 @@ void valkeyAsyncHandleTimeout(valkeyAsyncContext *ac) {
     valkeyAsyncDisconnectInternal(ac);
 }
 
-/* Sets a pointer to the first argument and its length starting at p. Returns
- * the number of bytes to skip to get to the following argument. */
-static const char *nextArgument(const char *start, const char **str, size_t *len) {
-    const char *p = start;
+/* Find the next argument in a command buffer, i.e. find the next bulkstring
+ * in an array of bulkstrings.
+ * Returns a pointer to the end of a found argument, which can be used when
+ * finding following arguments, or NULL when an argument is not found.
+ * The found string is returned by pointer via `str` and length in `strlen`. */
+static const char *nextArgument(const char *buf, size_t buflen, const char **str, size_t *strlen) {
+    const char *p = buf;
+    const size_t minlen = 7; /* '$1\r\n \r\n'*/
```

Comment:
Are we sure it can't be zero here?

```suggestion
    const size_t minlen = 6; /* '$0\r\n\r\n'*/
```

Reply:
Also this is personal preference but not all compilers will constant fold 

```c
const int minlen = 6
```

So a constant could be used instead:

```c
#define MIN_BULK_LEN ((size_t)6)
```

Reply:
I removed the check now, using memchr and the length parser made it much easier